nifi-nar-bundles/nifi-html-bundle/nifi-html-processors/src/main/java/org/apache/nifi/GetHTMLElement.java

```diff
@@ -0,0 +1,243 @@
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+        descriptors.add(URL);
+        descriptors.add(CSS_SELECTOR);
+        descriptors.add(HTML_CHARSET);
+        descriptors.add(OUTPUT_TYPE);
+        descriptors.add(DESTINATION);
+        descriptors.add(PREPEND_ELEMENT_VALUE);
+        descriptors.add(APPEND_ELEMENT_VALUE);
+        descriptors.add(ATTRIBUTE_KEY);
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_ORIGINAL);
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+        relationships.add(REL_NOT_FOUND);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final FlowFile flowFile = session.get();
+        if ( flowFile == null ) {
+            return;
+        }
+
+        try {
+
+            final Document doc = parseHTMLDocumentFromFlowfile(flowFile, context, session);
+            final Elements eles = doc.select(context.getProperty(CSS_SELECTOR)
+                    .evaluateAttributeExpressions().getValue());
+            final String prependValue = context.getProperty(PREPEND_ELEMENT_VALUE)
+                    .evaluateAttributeExpressions(flowFile).getValue();
+            final String appendValue = context.getProperty(APPEND_ELEMENT_VALUE)
+                    .evaluateAttributeExpressions(flowFile).getValue();
+
+            if (eles == null || eles.size() == 0) {
+                //No element found
+                session.transfer(flowFile, REL_NOT_FOUND);
+            } else {
+                for (final Element ele : eles) {
+                    final FlowFile ff = session.create();
+
+                    switch (context.getProperty(DESTINATION).getValue()) {
+                        case DESTINATION_ATTRIBUTE:
+                            final FlowFile atFlowfile = session.putAttribute(ff, HTML_ELEMENT_ATTRIBUTE_NAME,
+                                    extractElementValue(
+                                            prependValue,
+                                            context.getProperty(OUTPUT_TYPE).getValue(),
+                                            appendValue,
+                                            ele,
+                                            context.getProperty(ATTRIBUTE_KEY).evaluateAttributeExpressions()
+                                                    .getValue()));
+                            session.getProvenanceReporter().create(atFlowfile);
```

Comment:
This should really be a FORK, not a CREATE, since we are deriving content from an input FlowFile. However, there's no need to emit a Provenance Event at all, since the session.create(flowFile) mentioned above will automatically generate the event for you.